activesupport/lib/active_support/tagged_logging.rb

```diff
@@ -28,10 +28,13 @@ module ActiveSupport
   # it easy to stamp log lines with subdomains, request ids, and anything else
   # to aid debugging of multi-user production applications.
   module TaggedLogging
-    module Formatter # :nodoc:
-      # This method is invoked when a log event occurs.
-      def call(severity, timestamp, progname, msg)
-        super(severity, timestamp, progname, tag_stack.format_message(msg))
+    class TagProcessor # :nodoc:
+      def call(msg, logger)
+        if logger.formatter.nil?
+          logger.formatter ||= Logger::SimpleFormatter.new
+        end
+
+        tag_stack.format_message(msg)
```

Comment:
`TaggedLogging::Formatter` used to be included in the formatter one will use to format their logs.
It gives a given application the option to swap the formatter for a custom one, and have, centralized in one place, all the code that influences how a log entry is formatter (including tagging).

After your PR is merged, what option will I have, in my application, to change the format of my log entries from
```
[TAG1] [TAG2] This is my log entry
```
to, for example,
```
TAG1,TAG2 - This is my log entry
```
?

Reply:
Update: I just realized that `TaggedLogging::Formatter` is `:nodoc:` (ie. private API?)

Anyway, I actually tried to achieve what I claimed in my previous comment: use a custom formatter to change how tags get output. It's not straightforward, it relies on another `:nodoc:` API, but it works.

<details><summary>Check this snuppet.</summary>

```rb
class MyLoggerFormatter < ActiveSupport::Logger::SimpleFormatter
  include ActiveSupport::TaggedLogging::Formatter

  def tag_stack
    @thread_key ||= "activesupport_tagged_logging_tags:#{object_id}"
    ActiveSupport::IsolatedExecutionState[@thread_key] ||= TagStack.new
  end

  class TagStack < ActiveSupport::TaggedLogging::TagStack
    def format_message(message)
      if @tags.empty?
        message
      else
        "#{@tags.join(",")} - #{message}"
      end
    end
  end
end

logger = ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new($stdout)
logger.formatter = MyLoggerFormatter.new

logger.tagged("TAG1") do
  logger.tagged("TAG2") do
    logger.info("This is a test")
  end
end

# Output:
TAG1,TAG2 - This is a test
```

</details> 

If custom formatters are an (un)officially supported Rails API, we might want to consider it anyway...